client/packages/programs/src/JsonForms/components/Search/useSearchQueries.tsx

```diff
@@ -1,69 +1,55 @@
-import React, { useState } from 'react';
-import {
-  Typography,
-  useTranslation,
-  RegexUtils,
-} from '@openmsupply-client/common';
-import { usePatient } from '@openmsupply-client/system';
+import { useState } from 'react';
+import { RegexUtils, FilterBy } from '@openmsupply-client/common';
+import { PatientRowFragment, usePatient } from '@openmsupply-client/system';
 
-export const QueryValues = ['patientByCode'] as const;
+export const QueryValues = ['patientSearch'] as const;
 type QueryValue = (typeof QueryValues)[number];
 
-type GetDisplayElement = (
-  result: Record<string, unknown>
-) => JSX.Element | null;
-
 interface SearchQueryOptions {
   query?: QueryValue;
   optionString?: string;
   displayString?: string;
   saveFields?: string[];
-  placeholderText?: string;
 }
 
 interface SearchQueryOutput {
-  runQuery: (searchValue: string) => void;
+  runQuery: (searchFilter: FilterBy) => void;
   saveFields: string[] | null;
-  getOptionLabel?: (result: Record<string, unknown>) => string;
-  getDisplayElement?: GetDisplayElement;
-  placeholderText: string;
+  getOptionLabel: (result: PatientRowFragment) => string;
 }
 
-const { formatTemplateString, removeEmptyLines } = RegexUtils;
+const { formatTemplateString } = RegexUtils;
 
 export const useSearchQueries = ({
   query,
   optionString,
-  displayString,
   saveFields,
-  placeholderText,
 }: SearchQueryOptions = {}) => {
-  const [results, setResults] = useState<Record<string, unknown>[]>([]);
+  const [results, setResults] = useState<PatientRowFragment[]>([]);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(false);
 
-  const t = useTranslation('programs');
-
   const patientQueries = usePatient.utils.api();
 
   const searchQueries: Record<QueryValue, SearchQueryOutput> = {
-    patientByCode: {
-      runQuery: async (searchValue: string) => {
-        if (searchValue === '') {
-          setResults([]);
-          return;
-        }
+    patientSearch: {
+      runQuery: async (searchFilter: FilterBy) => {
         setError(false);
         setLoading(true);
         patientQueries.get
           .list({
             first: 10,
             offset: 0,
             sortBy: { key: 'lastName', direction: 'asc' },
-            filterBy: { code: { like: searchValue } },
+            filterBy: searchFilter,
           })
           .then(result => {
-            setResults(result.nodes);
+            setResults(
+              // If patient has a full document field, use that since it'll make
+              // more data available. Otherwise just use the basic Patient
+              // fields
+              result.nodes.map(patient => patient.document?.data ?? patient)
```

Comment:
Nit: you could add the expected document data type for the Patient to `packages/programs/src/schema_types.ts` (at least a minimal PatientSchema type that has an overlap with the PatientRowFragment). By using this you can make sure you only use fields that exist in both types...

Reply:
I'm not sure what you mean exactly -- how/where would this be enforced?

Also -- the purpose of using the "document" version if available is that it would be possible to save *non*-standard Patient fields if they existed in the Patient schema. (The way the "handlePatientSelect" function currently works, it will just silently ignore any defined "saveFields" that don't exist on the results, so won't cause any problems if saveFields are defined that don't exist on some patients)

Reply:
I mean `results` has a defined type of `PatientRowFragment[]` but its either the document data or the PatientRowFragment, i.e. the type is wrong. So bit hard to reason about what is actually be returned. Might also be confusing to read...